lld/ELF/Arch/RISCV.cpp

```diff
@@ -1084,10 +1084,63 @@ static void mergeArch(RISCVISAInfo::OrderedExtensionMap &mergedExts,
   }
 }
 
+static void mergeAtomic(DenseMap<unsigned, unsigned> &intAttr,
+                        const InputSectionBase *oldSection,
+                        const InputSectionBase *newSection, unsigned int oldTag,
+                        unsigned int newTag) {
+  using RISCVAttrs::RISCVAtomicAbiTag::AtomicABI;
+  // Same tags stay the same, and UNKNOWN is compatible with anything
+  if (oldTag == newTag || newTag == AtomicABI::UNKNOWN)
+    return;
+
+  auto attr = RISCVAttrs::ATOMIC_ABI;
+  switch (oldTag) {
+  case AtomicABI::UNKNOWN:
+    intAttr[attr] = newTag;
```

Comment:
Can we use the original iterator returned by try_emplace to avoid the second map lookup here?

Reply:
done.